Sources/PackageLoading/ManifestLoader.swift

```diff
@@ -768,70 +628,47 @@ public final class ManifestLoader: ManifestLoaderProtocol {
-
-                let vfsOverlayTempFilePath = tempDir.appending("vfs.yaml")
-                try VFSOverlay(roots: [
-                    VFSOverlay.File(
-                        name: manifestPath._normalized.replacing(#"\"#, with: #"\\"#),
-                        externalContents: manifestTempFilePath._nativePathString(escaped: true)
-                    )
-                ]).write(to: vfsOverlayTempFilePath, fileSystem: localFileSystem)
+        return try await Basics.withTemporaryDirectory(removeTreeOnDeinit: true) { tempDir in
+            let manifestTempFilePath = tempDir.appending("manifest.swift")
+            // Since this isn't overwriting the original file, append Foundation library
+            // import to avoid having diagnostics being displayed on the incorrect line.
+            try localFileSystem.writeFileContents(manifestTempFilePath, bytes: ByteString(manifestContents + manifestPreamble.contents))
+
+            let vfsOverlayTempFilePath = tempDir.appending("vfs.yaml")
+            try VFSOverlay(roots: [
+                VFSOverlay.File(
+                    name: manifestPath._normalized.replacing(#"\"#, with: #"\\"#),
+                    externalContents: manifestTempFilePath._nativePathString(escaped: true)
+                )
+            ]).write(to: vfsOverlayTempFilePath, fileSystem: localFileSystem)
 
-                validateImports(
-                    manifestPath: manifestTempFilePath,
-                    toolsVersion: toolsVersion,
-                    callbackQueue: callbackQueue
-                ) { result in
-                    dispatchPrecondition(condition: .onQueue(callbackQueue))
-
-                    do {
-                        try result.get()
-
-                        try self.evaluateManifest(
-                            at: manifestPath,
-                            vfsOverlayPath: vfsOverlayTempFilePath,
-                            packageIdentity: packageIdentity,
-                            packageLocation: packageLocation,
-                            toolsVersion: toolsVersion,
-                            observabilityScope: observabilityScope,
-                            delegate: delegate,
-                            delegateQueue: delegateQueue,
-                            callbackQueue: callbackQueue
-                        ) { result in
-                            dispatchPrecondition(condition: .onQueue(callbackQueue))
-                            cleanupTempDir(tempDir)
-                            completion(result)
-                        }
-                    } catch {
-                        cleanupTempDir(tempDir)
-                        callbackQueue.async {
-                            completion(.failure(error))
-                        }
-                    }
-                }
-            }
-        } catch {
-            callbackQueue.async {
-                completion(.failure(error))
-            }
+            let _ = try await validateImports(
```

Comment:
what's the reason for `let _ =` if this function has no return value in the first place?
```suggestion
            try await validateImports(
```